### ISSUE TITLE
[Bug]: Able to write a TimeSeries with zero rate

### ISSUE THREAD

**Issue by CodyCBakerPhD** [NeurodataWithoutBorders/pynwb#1721]
### What happened?

I'm able to run the following code successfully - whereas it seems to me that should be prohibited.

Will definitely add a check for this on the Inspector side, but what are your thoughts @rly @oruebel?

### Steps to Reproduce

```python
from pynwb.testing.mock.base import mock_TimeSeries

mock_TimeSeries(rate=0.0)
```

### Operating System

Windows

### Python Executable

Conda

### Python Version

3.11

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/.github/CODE_OF_CONDUCT.rst)
- [X] Have you checked the [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured this bug was not already [reported](https://github.com/NeurodataWithoutBorders/pynwb/issues)?

**Comment by bendichter** [comment]
I wouldn't be surprised if <0 also works without error. On the road right now or I'd test it

**Comment by CodyCBakerPhD** [comment]
> I wouldn't be surprised if <0 also works without error. On the road right now or I'd test it

```python
from pynwb.testing.mock.base import mock_TimeSeries

mock_TimeSeries(rate=-1.0)
```

also worked, at least in memory. Haven't tried calling `io.write` yet

**Comment by oruebel** [comment]
I think PyNWB only checks for the data type but does not restrict the value range. Adding a check for this in the TimeSeries constructor and at least raising a warning seems reasonable. If we want to raise an error, then we need to distinguish between read/create mode to ensure that files can be read (i.e., warn on read and raise ValueError when creating a new TimeSeries).

**Comment by oruebel** [comment]
> distinguish between read/create mode

I vaguely remember that we had added logic to support this some time ago, but will need to look at the code because I don't remember off-the-top-of-my-head where this logic is.

**Comment by bendichter** [comment]
@rly you added that logic, right?

**Comment by rly** [comment]
Fixed in #1793
